### ISSUE TITLE
Copy link and copy source buttons in snippet details

### ISSUE THREAD

**Issue by tpetricek** [tpetricek/FsSnip.Website#9]
... are not implemented! They should copy the thing into clipboard (if we can do that without ugly hacks - if no, we should remove the buttons)

**Comment by abjrcode** [comment]
I have been looking around for a way to do this smoothly but it seems there is no "clean" way to do it.

Google Chrome has removed all APIs that grant access the clipboard and there is no standard API for the others browsers too.

There are two hacks that people are using to work around this.
- You can either pop up a dialog containing the required text already selected and ask the user to print Ctrl-C or any other shortcut of course.
- The other work around is injecting a flash script which grants you access to the clipboard. This is the popular solution because it stands out of the users way and it is used by GitHub itself when you copy a repository's URL with the copy button. They are using a famous third party library for this called **ZeroClipboard**.

Let me know what you think Tomas.

**Comment by tpetricek** [comment]
Yeah, the current version of www.fssnip.net uses the Flash trick, but I don't think that's a very good solution. I quite like the idea of just doing a pop up dialog (perhaps some nice Bootstrap way) with a textbox where people can copy it, so I'd go for that option!

**Comment by abjrcode** [comment]
Okay. I am on it.

**Comment by abjrcode** [comment]
I am almost done but I honestly don't like the way things came up at last.  
Anyway I will create a pull request soon and you could see for your self.

**Comment by tpetricek** [comment]
This works in Chrome, but it seems to be broken in IE (on my machine at least).

Also, I'm not particularly keen on the blueish label in the dialog, so if we can make that white, that would be nicer (I think!)

**Comment by tpetricek** [comment]
I couldn't figure out how to make this work in IE, so I added exception handler...

**Comment by abjrcode** [comment]
Oh sorry I totally forgot about this in the mid of all the work I have, but yes I remember that accessing the clipboard in a uniform way across browsers is really hard. I miss F# and I hope to be able to participate again although I can't see it happening soon. Good luck

**Comment by kunjee17** [comment]
@tpetricek have a look at [clipboard.js](https://zenorocha.github.io/clipboard.js/) whenever you have time. It is having decent browser support.
